# Dampak Penggunaan To-Do-List dalam Peningkatan Produktivitas

Source Code: Muhammad Naufal Faza

Dokumentasi: Michael Harditya

In [1]:
!pip install pandas scipy

  Using cached scipy-1.14.1-cp312-cp312-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (60 kB)
  Using cached tzdata-2024.1-py2.py3-none-any.whl.metadata (1.4 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 12.7/12.7 MB 8.0 MB/s eta 0:00:00a 0:00:01
Using cached scipy-1.14.1-cp312-cp312-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (40.8 MB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 16.0/16.0 MB 5.7 MB/s eta 0:00:00a 0:00:01
Using cached tzdata-2024.1-py2.py3-none-any.whl (345 kB)


## Load Data
Data yang digunakan dapat diakses pada folder `data`.

Terdapat tiga file didalamnya yaitu `before_management.csv` yang berisi data sebelum penggunaan to-do-list,
`after_management.csv` dan `after_management_2.csv` yang berisi data sesudah penggunaan to-do-list.

In [15]:
import pandas as pd

# Load CSV
before_management = pd.read_csv('data/before_management.csv')
after_management_1 = pd.read_csv('data/after_management.csv')
after_management_2 = pd.read_csv('data/after_management_2.csv')

## Preprocessing
Data digabungkan (khususnya pada `after_management`) dan kemudian disesuaikan formatnya agar dapat dihitung oleh program.

### Gabung `after_management` dengan `after_management_2`

In [55]:
# menggabungkan after_management_1 dan after_management_2
after_management = pd.concat([after_management_1, after_management_2])

In [27]:
before_management.describe(include='all')

payableDate                    itemID                   duration  \
count             11                        11                         11   
unique             4                        11                        NaN   
top     Aug 12, 2024  66b9a3d305d8b945184163c4                        NaN   
freq               3                         1                        NaN   
mean             NaN                       NaN  0 days 00:39:37.363636363   
std              NaN                       NaN  0 days 00:13:33.003723574   
min              NaN                       NaN            0 days 00:00:00   
25%              NaN                       NaN     0 days 00:41:33.500000   
50%              NaN                       NaN            0 days 00:43:02   
75%              NaN                       NaN            0 days 00:43:11   
max              NaN                       NaN            0 days 00:53:08   

       rateApplied  payout payType        projectName     status  
count           11      11      11                 11         11  
unique           2       9       2                  2          1  
top         $19.50  $14.03  prepay  Article Orangutan  processed  
freq            10       2      10                 10         11  
mean           NaN     NaN     NaN                NaN        NaN  
std            NaN     NaN     NaN                NaN        NaN  
min            NaN     NaN     NaN                NaN        NaN  
25%            NaN     NaN     NaN                NaN        NaN  
50%            NaN     NaN     NaN                NaN        NaN  
75%            NaN     NaN     NaN                NaN        NaN  
max            NaN     NaN     NaN                NaN        NaN

In [28]:
after_management.describe(include='all')

payableDate                    itemID                   duration  \
count             48                        48                         48   
unique            10                        46                        NaN   
top     Sep 19, 2024  66e838199071dcf88a46ee4c                        NaN   
freq              14                         2                        NaN   
mean             NaN                       NaN  0 days 00:35:08.083333333   
std              NaN                       NaN  0 days 00:19:47.753589197   
min              NaN                       NaN            0 days 00:00:00   
25%              NaN                       NaN     0 days 00:24:04.750000   
50%              NaN                       NaN            0 days 00:29:48   
75%              NaN                       NaN     0 days 00:49:14.500000   
max              NaN                       NaN            0 days 01:22:07   

       rateApplied  payout payType                        projectName  \
count           48      48      48                                 48   
unique           5      35       5                                  3   
top         $19.50  $25.00  prepay  Constellation SFT Math Singleturn   
freq            25       4      25                                 23   
mean           NaN     NaN     NaN                                NaN   
std            NaN     NaN     NaN                                NaN   
min            NaN     NaN     NaN                                NaN   
25%            NaN     NaN     NaN                                NaN   
50%            NaN     NaN     NaN                                NaN   
75%            NaN     NaN     NaN                                NaN   
max            NaN     NaN     NaN                                NaN   

           status  
count          48  
unique          2  
top     processed  
freq           26  
mean          NaN  
std           NaN  
min           NaN  
25%           NaN  
50%           NaN  
75%           NaN  
max           NaN

### Konversi data waktu

In [17]:
# konversi kolom duration ke format `timedelta`
def convert_duration(duration_str):
    if pd.isna(duration_str) or duration_str == "-":
        return pd.Timedelta(0)
    try:
        return pd.to_timedelta(duration_str)
    except ValueError:
        # error handler
        return pd.Timedelta(0)

before_management['duration'] = before_management['duration'].apply(convert_duration)
after_management['duration'] = after_management['duration'].apply(convert_duration)

### Grouping data per hari

In [18]:
# Melakukan grouping berdasarkan `payableDate` dan menjumlahkan durasi per hari
before_grouped = before_management.groupby('payableDate')['duration'].sum().reset_index()
after_grouped = after_management.groupby('payableDate')['duration'].sum().reset_index()

## Statistical Testing
Melakukan dua pengujian yaitu pengujian *normality* menggunakan uji Shapiro-Wilk, dan pengujian hipotesis.

### Shapiro-Wilk: Uji *normality*
Fungsi dapat direferensikan pada paper.

In [19]:
from scipy.stats import shapiro

# Melakukan uji Shapiro-Wilk
shapiro_before = shapiro(before_grouped['duration'])
shapiro_after = shapiro(after_grouped['duration'])

print("Shapiro-Wilk Test - Before Management:", shapiro_before)
print("Shapiro-Wilk Test - After Management:", shapiro_after)

Shapiro-Wilk Test - Before Management: ShapiroResult(statistic=np.float64(0.8303459008610953), pvalue=np.float64(0.16870582277438945))
Shapiro-Wilk Test - After Management: ShapiroResult(statistic=np.float64(0.8394128841296956), pvalue=np.float64(0.043415258287793496))


In [24]:
before_grouped.describe(include='all')

payableDate                   duration
count              4                          4
unique             4                        NaN
top     Aug 10, 2024                        NaN
freq               1                        NaN
mean             NaN     0 days 01:48:57.750000
std              NaN  0 days 00:22:58.609776792
min              NaN            0 days 01:24:51
25%              NaN     0 days 01:31:33.750000
50%              NaN     0 days 01:50:48.500000
75%              NaN     0 days 02:08:12.500000
max              NaN            0 days 02:09:23

In [26]:
after_grouped.describe(include='all')

payableDate                   duration
count             10                         10
unique            10                        NaN
top     Sep 11, 2024                        NaN
freq               1                        NaN
mean             NaN     0 days 02:48:38.800000
std              NaN  0 days 01:59:47.102550170
min              NaN            0 days 00:49:08
25%              NaN     0 days 00:59:29.750000
50%              NaN     0 days 02:19:46.500000
75%              NaN     0 days 04:43:33.500000
max              NaN            0 days 05:35:29

Nilai tingkat signifikansi alpha yang digunakan adalah `0.05`.

Pada kondisi "sebelum" penggunaan to-do-list terlihat bahwa nilai `p-value` sebesar `0.1687`, disimpulkan bahwa data "sebelum" terdistribusi secara normal.

Sedangkan pada kondisi "sesudah" penggunaan to-do-list terlihat bahwa nilai `p-value` sebesar `0.0434`, disimpulkan bahwa data "sesudah" tidak terdistribusi secara normal.

Karena itu, tes hipotesis akan dilanjutkan dengan metode Mann-Whitney.

### Mann-Whitney: Uji Hipotesis

Fungsi mereferensikan pada paper.

Hipotesis yang akan diuji:

-	H0 (hipotesis nol): Tidak ada perbedaan signifikan dalam durasi jam kerja harian sebelum dan sesudah penggunaan to-do list (Google Calendar).
-	H1 (hipotesis alternatif): Ada perbedaan signifikan dalam durasi jam kerja harian sebelum dan sesudah penggunaan to-do list (Google Calendar).

In [10]:
from scipy.stats import mannwhitneyu

# Melakukan uji Mann-Whitney U
u_stat, p_val = mannwhitneyu(before_grouped['duration'], after_grouped['duration'])

print(f"Mann-Whitney U Test result: U-statistic={u_stat}, p-value={p_val}")


Mann-Whitney U Test result: U-statistic=18.0, p-value=0.8391608391608392


Dengan tingkat signifikansi alpha `0.05`.

Nilai `p-value` yang didapatkan adalah `0.839`, sehingga kesimpulannya:

**H0 gagal ditolak, berdasarkan data ynag dianalisis, tidak cukup bukti untuk menyatakan bahwa terdapat perbedaan yang signifikan dalam distribusi durasi jam kerja harian antara kondisi "sebelum" dan "sesudah" penerapan to-do-list.**